### PR TITLE
feat(Datalist/TMC-18150): add icons to datalist items

### PR DESCRIPTION
**Pull request by yonmey**
**What is the problem this PR is trying to solve?**
Sometimes we'd want to add a visual element, an icon, next to the Datalist items. This was asked in https://jira.talendforge.org/browse/TMC-18150.

**What is the chosen solution to this problem?**
I added the possibility to pass an `icon` object props to be placed on the left of each datalist's item:

image

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/Datalist/Datalist.component.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ class Datalist extends Component {
 		this.state = {
 			previousValue: props.value,
 			value: props.value,
+			selecteditem: props.value,
```
Comment (sgendre) [inline packages/components/src/Datalist/Datalist.component.js]: camelCase please

File: packages/components/src/Typeahead/Typeahead.component.renderers.js [hunk 2]
```diff
@@ -35,6 +36,7 @@ export function renderInputComponent(props) {
 			<ControlLabel srOnly htmlFor={key}>
 				Search
 			</ControlLabel>
+			{get(props, 'selecteditem.icon') && <Icon {...props.selecteditem.icon} className={theme['item-icon']} />}
```
Comment (sgendre) [inline packages/components/src/Typeahead/Typeahead.component.renderers.js]: inverting className props and the spreading would alllow className customization through props

File: packages/components/stories/Datalist.js [hunk 3]
```diff
@@ -54,6 +54,18 @@ storiesOf('Datalist', module)
 	.add('default multiSection', () => {
 		const restrictedValues = { ...propsMultiSection, restricted: true };
 		const defaultValue = { ...propsMultiSection, value: 'lol' };
+		const withIcons = {
+			...propsMultiSection,
+			titleMap: propsMultiSection.titleMap.map(tm => ({
```
Comment (sgendre) [inline packages/components/stories/Datalist.js]: please avoid acronyms

File: packages/components/stories/Datalist.js [hunk 4]
```diff
@@ -54,6 +54,18 @@ storiesOf('Datalist', module)
...
+				...tm,
+				suggestions: tm.suggestions.map(
+					s => ({
```
Comment (sgendre) [inline packages/components/stories/Datalist.js]: same here

File: packages/components/stories/Datalist.js [hunk 5]
```diff
@@ -72,6 +86,16 @@ storiesOf('Datalist', module)
 		const restrictedValues = { ...singleSectionProps, restricted: true };
 		const defaultValue = { ...singleSectionProps, value: 'lol' };
 		const disabledItems = { ...singleSectionProps, titleMap: titleMapWithDisabledItems };
+		const withIcons = {
+			...singleSectionProps,
+			titleMap: singleSectionProps.titleMap.map((tm, i) => ({
```
Comment (sgendre) [inline packages/components/stories/Datalist.js]: and here

File: packages/components/src/Datalist/Datalist.component.js [hunk 6]
```diff
@@ -202,6 +203,30 @@ class Datalist extends Component {
...
+	 */
+	getSelectedIcon() {
+		if (this.props.titleMap) {
+			const multiSection = this.props.titleMap.find(
+				titleMap =>
+					titleMap.suggestions &&
+					titleMap.suggestions.find(suggestion => suggestion.name === this.state.value),
+			);
+
+			if (multiSection) {
```
Comment (jmfrancois) [inline packages/components/src/Datalist/Datalist.component.js]: you can use props.multiselection

**Pull request merged**